Build failures for ucans32k1sic/s32k146

**Describe the bug**

Hi NXP folks, seems like ucans32k1sic does not build correctly on most samples (any one that uses gpio it seems)

```
$ west build -p -b ucans32k1sic/s32k146 samples/basic/blinky
...
zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/gpio/fsl_gpio.c:29:41: error: 'PORT_BASE_PTRS' undeclared here (not in a function); did you mean 'RTC_BASE_PTRS'?
   29 | static PORT_Type *const s_portBases[] = PORT_BASE_PTRS;
      |                                         ^~~~~~~~~~~~~~
      |                                         RTC_BASE_PTRS
zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/gpio/fsl_gpio.c:30:41: error: 'GPIO_BASE_PTRS' undeclared here (not in a function); did you mean 'LPI2C_BASE_PTRS'?
   30 | static GPIO_Type *const s_gpioBases[] = GPIO_BASE_PTRS;
      |                                         ^~~~~~~~~~~~~~
      |                                         LPI2C_BASE_PTRS
```

Looks like it broke on one of the hal updates, 87579e8c788afa363747366934d252a3cdaa7a3c specifically.

Is this something you could look into? Accounts for quite few failures in the weekly run.

**To Reproduce**
See above

**Expected behavior**
No errors

**Impact**
Probably the whole platform is unusable

**Logs and console output**
See above

**Environment (please complete the following information):**

Linux, Zephyr SDK 0.16.1, baced6d315f

--- a/drivers/gpio/gpio_mcux.c
+++ b/drivers/gpio/gpio_mcux.c
@@ -1,6 +1,6 @@
 /*
  * Copyright (c) 2016 Freescale Semiconductor, Inc.
- * Copyright 2017, 2023-2024 NXP
+ * Copyright 2017, 2023-2025 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
@@ -19,9 +19,6 @@
 
 #if defined(CONFIG_PINCTRL_NXP_IOCON)
 #include <fsl_iopctl.h>
-#include <fsl_reset.h>
-#include <fsl_clock.h>
-#include <fsl_gpio.h>
 /* Use IOCON to configure electrical characteristic, set PORT_Type as void. */
 #define PORT_Type void
 #endif

--- a/west.yml
+++ b/west.yml
@@ -203,7 +203,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 49ff7e33f848e4b59da59369a77da63e346fb1a3
+      revision: 0b86efa2ad563e60d4d57fe7fa7ac6ccfa092f8c
       path: modules/hal/nxp
       groups:
         - hal

